fdbserver/storageserver.actor.cpp

```diff
@@ -8724,12 +8759,80 @@ bool fetchKeyCanRetry(const Error& e) {
 	case error_code_commit_proxy_memory_limit_exceeded:
 	case error_code_storage_replica_comparison_error:
 	case error_code_unreachable_storage_replica:
+	case error_code_bulkload_task_failed: // for fetchKey based bulkload
 		return true;
 	default:
 		return false;
 	}
 }
 
+ACTOR Future<BulkLoadFileSet> bulkLoadFetchKeyValueFileToLoad(StorageServer* data,
+                                                              std::string dir,
+                                                              BulkLoadTaskState bulkLoadTaskState) {
+	ASSERT(bulkLoadTaskState.getLoadType() == BulkLoadType::SST);
+	TraceEvent(SevInfo, "SSBulkLoadTaskFetchSSTFile", data->thisServerID)
+	    .setMaxEventLength(-1)
+	    .setMaxFieldLength(-1)
+	    .detail("BulkLoadTask", bulkLoadTaskState.toString())
+	    .detail("Dir", abspath(dir));
+	state double fetchStartTime = now();
+	// Download data file from fromRemoteFileSet to toLocalFileSet
+	state BulkLoadFileSet fromRemoteFileSet = bulkLoadTaskState.getFileSet();
+	state BulkLoadFileSet toLocalFileSet = wait(bulkLoadDownloadTaskFileSet(
+	    bulkLoadTaskState.getTransportMethod(), fromRemoteFileSet, dir, data->thisServerID));
+	// Do not need byte sampling locally in fetchKeys
+	const double duration = now() - fetchStartTime;
```

Comment:
nit: we could use a histogram to track latency here

Reply:
I will do a separate PR for latency measurement.